src/ruby/puppetserver-lib/puppet/server/ast_compiler.rb

```diff
@@ -0,0 +1,53 @@
+require 'puppet_pal'
+require 'puppet/server/log_collector'
+require 'puppet/server/logging'
+
+module Puppet
+  module Server
+    class ASTCompiler
+      def self.compile(compile_options)
+        options = compile_options['options'] || {}
+
+        log_level = options['log_level']
+        code = JSON.parse(compile_options['code_ast'])
+
+        if options['capture_logs']
+          catalog, logs = Logging.capture_logs(log_level) do
+            compile_ast(code, compile_options)
+          end
+
+          { catalog: catalog, logs: logs }
+        else
+          catalog = compile_ast(code, compile_options)
+          { catalog: catalog }
+        end
+      end
+
+      def self.compile_ast(code, compile_options)
+        target = compile_options['certname']
```

Comment:
It looks like `target` isn't used anywhere else in this method? 🎯

Reply:
True... looks like Bolt might set `Puppet[:node_name_value]` to the cert name, I didn't seem to need that, but maybe you do in some cases...

Reply:
Updated to do what Bolt does. PAL uses it notably here: https://github.com/puppetlabs/puppet/blob/master/lib/puppet/pal/pal_impl.rb#L320 so setting that seems correct.